src/Storages/MergeTree/MergeTreeIndexBloomFilter.cpp

```diff
@@ -325,6 +376,60 @@ bool MergeTreeIndexConditionBloomFilter::mayBeTrueOnGranule(const MergeTreeIndex
     return rpn_stack[0].can_be_true;
 }
 
+bool MergeTreeIndexConditionBloomFilter::isIndexUseful() const
+{
+    std::vector<RPNEvaluationIndexUsefulnessState> rpn_stack;
+    rpn_stack.reserve(rpn.size() - 1);
+
+    for (const auto & element : rpn)
```

Comment:
The new function duplicates (or at least overlaps) with what `alwaysUnknownOrTrue` is doing. I guess we should unify both.

Reply:
3b48a692c6bbb2c01be90ba789cc3d3187529b22